src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -12,19 +1,72 @@
+import React from 'react';
+import { Filters } from '../../types/Filters';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+const filters = ['All', 'Active', 'Completed'];
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+type Props = {
+  handleFilterChange: React.Dispatch<React.SetStateAction<Filters>>;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  query: string;
+  selectedFilter: Filters;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  handleFilterChange,
+  setQuery,
+  query,
+  selectedFilter,
+}) => {
+  const onFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    handleFilterChange(event.target.value as Filters);
+  };
+
+  const onInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value.trimStart());
+  };
+
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={selectedFilter}
+            onChange={onFilterChange}
+          >
+            {filters.map(filter => (
+              <option value={filter.toLowerCase()} key={filter}>
```

Comment:
The `value` attribute for the `<option>` elements is set to `filter.toLowerCase()`, which may not match the `Filters` enum values. Ensure that the values here correspond to the enum values to avoid issues in the `onFilterChange` function.